### PR TITLE
[gtest] Update to version 1.8.1

### PR DESCRIPTION
**Pull request by KindDragon**

### CODE REVIEW

File: ports/gtest/portfile.cmake [hunk 1]
```diff
@@ -50,35 +48,30 @@ file(
...
-if(EXISTS ${CURRENT_PACKAGES_DIR}/debug/lib/gtestd.lib)
-    file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link)
-    file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/gtestd.lib ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/gtestd.lib)
-    file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/gtest_maind.lib ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/gtest_maind.lib)
-    file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/gmockd.lib ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/gmockd.lib)
-    file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/gmock_maind.lib ${CURRENT_PACKAGES_DIR}/debug/lib/manual-link/gmock_maind.lib)
+
+    file(READ ${CURRENT_PACKAGES_DIR}/share/gtest/GTestTargets-release.cmake RELEASE_CONFIG)
+    string(REPLACE "\${_IMPORT_PREFIX}/lib/gtest_main.lib"
+                   "\${_IMPORT_PREFIX}/manual-link/gtest_main.lib" RELEASE_CONFIG "${RELEASE_CONFIG}")
```
Comment (Perlmint) [inline ports/gtest/portfile.cmake]: Hi, This PR is what I just needed, so I used your patch of this PR.
It almost works well, but finding running CMake for MSVS to find GTest failed with file not found error.
I tried to fix error, finally I found typo on this line.

**manual-link/gtest_main.lib** is placed in **lib/manual-link/gtest_main.lib**, **lib/** is missing here. and [Line number 73](https://github.com/Microsoft/vcpkg/pull/4220/files#diff-82d6fc85ad49ab0955f34727c25e73a2R73) has similar typo.

After fixing these lines, It works nicely.

**Review by Florian-Reiser — APPROVED** [review]
I've tested the upgrade to gtest 1.8.1 and I've encountered no problems so far.

**Comment by rleigh-codelibre** [comment]
Should the 1.8.1 update also add the fixes from master for https://github.com/google/googletest/issues/1111 which are required to work with C++17 with both VS2017 and VS2015?

File: ports/gtest/usage [hunk 2]
```diff
@@ -2,5 +2,5 @@ The package gtest is compatible with built-in CMake targets:
 
     enable_testing()
     find_package(GTest REQUIRED)
-    target_link_libraries(main PRIVATE GTest::GTest GTest::Main)
+    target_link_libraries(main PRIVATE GTest::gtest_main) # or GTest::gmock_main
```
Comment (rleigh-codelibre) [inline ports/gtest/usage]: Why the change here?  `GTest::GTest`, `GTest::Main`, are the documented targets to use in the cmake module documentation.

**Comment by ras0219-msft** [comment]
Thanks a bunch for this PR and sorry for the long delay!

Bonus thanks as well to @rleigh-codelibre and @Florian-Reiser for trying it out; this really helps as well!

I followed @rleigh-codelibre's suggestion and reverted the targets back to the official cmake[1] recommendation. I think that before we change our displayed targets here, it should be changed upstream in cmake.

[1] https://cmake.org/cmake/help/latest/module/FindGTest.html
